compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -1949,72 +1952,99 @@ void startFromClause() {
+                                                       boolean isDeclaredWithVar, boolean isFromClause) {
         BLangTupleVariableDef variableDefinitionNode = createTupleVariableDef(pos, ws, false,
                 false, isDeclaredWithVar);
-        if (!this.bindingPatternIdentifierWS.isEmpty() && !isInQuery) {
+        if ((!this.bindingPatternIdentifierWS.isEmpty() && !isInQuery) || !this.bindingPatternIdentifierWS.isEmpty()) {
             variableDefinitionNode.addWS(this.bindingPatternIdentifierWS.pop());
         } else if (!this.queryBindingPatternIdentifierWS.isEmpty()) {
             variableDefinitionNode.addWS(this.queryBindingPatternIdentifierWS.pop());
         }
 
-        isInQuery = false;
-        addFromClause(pos, ws, variableDefinitionNode, isDeclaredWithVar);
+        if (isFromClause) {
+            isInQuery = false;
+        }
+        addClause(pos, ws, variableDefinitionNode, isDeclaredWithVar, isFromClause);
     }
 
-    private void addFromClause(DiagnosticPos pos, Set<Whitespace> ws,
-                               VariableDefinitionNode variableDefinitionNode, boolean isDeclaredWithVar) {
-        BLangFromClause fromClause = (BLangFromClause) TreeBuilder.createFromClauseNode();
-        fromClause.addWS(ws);
-        fromClause.pos = pos;
-        markVariableAsFinal((BLangVariable) variableDefinitionNode.getVariable());
-        fromClause.setVariableDefinitionNode(variableDefinitionNode);
-        fromClause.setCollection(this.exprNodeStack.pop());
-        fromClause.isDeclaredWithVar = isDeclaredWithVar;
-        queryClauseStack.push(fromClause);
+    private void addClause(DiagnosticPos pos, Set<Whitespace> ws,
+                               VariableDefinitionNode variableDefinitionNode, boolean isDeclaredWithVar,
+                           boolean isFromClause) {
+        if (isFromClause) {
+            BLangFromClause fromClause = (BLangFromClause) TreeBuilder.createFromClauseNode();
+            fromClause.addWS(ws);
+            fromClause.pos = pos;
+            markVariableAsFinal((BLangVariable) variableDefinitionNode.getVariable());
+            fromClause.setVariableDefinitionNode(variableDefinitionNode);
+            fromClause.setCollection(this.exprNodeStack.pop());
+            fromClause.isDeclaredWithVar = isDeclaredWithVar;
+            queryClauseStack.push(fromClause);
+        } else {
+            BLangJoinClause joinClause = (BLangJoinClause) TreeBuilder.createJoinClauseNode();
+            joinClause.addWS(ws);
+            joinClause.pos = pos;
+            markVariableAsFinal((BLangVariable) variableDefinitionNode.getVariable());
+            joinClause.setVariableDefinitionNode(variableDefinitionNode);
+            joinClause.isDeclaredWithVar = isDeclaredWithVar;
+            joinClause.setCollection(this.exprNodeStack.pop());
+            queryClauseStack.push(joinClause);
+        }
+
```

Comment:
unnecessary new line

Reply:
`BLangPackageBuilder.java` is no longer used